packages/core/src/provider/provider.ts

```diff
@@ -0,0 +1,11 @@
+// provider.ts
+import { EventEmitter } from "events";
+import { IRequestOptions } from "./types";
+
+interface IProvider {
```

Comment:
I wonder if provider should expose a similar interface to commands:

```
set_entity(..)
get_entity(..)
get_entities(..)
```

Reply:
I think we also want to expose `get_component`